### ISSUE TITLE
Single publisher to many subscribers

### ISSUE THREAD

**Issue by jbabyhacker** [jhelovuo/RustDDS#12]
I am messing around with the Shapes demo. I am able to publish a square from the eProsima gui, and the shapes demo that you wrote is able to receive it. However, if I have two shapes demo processes running, in the same domain to receive squares, only one is able to receive. The other process that is unable to receive prints the following error. Does the current RustDDS implementation support this topology?

```bash
ERROR rustdds::dds::participant] Cannot join multicast, possibly another instance running on this machine.
```

**Comment by nime88** [comment]
If i remember correctly this was/is limitation of rust std::net::UDPSocket, thus there is no clear support for two instances in same machine in the same domain. This is one of the issues that might need to be addressed later.

**Comment by jhelovuo** [comment]
> If i remember correctly this was/is limitation of rust std::net::UDPSocket, thus there is no clear support for two instances in same machine in the same domain. This is one of the issues that might need to be addressed later.

Yes and no.

There is (apparently) is no support for setting the SO_REUSEADDR socket option between socket() and bind() syscalls, but this can be circumvented by making the socket using the socket2 crate.

This should now be fixed in  RustDDS version 0.2.3. 

@jbabyhacker , please comment if it works for you.

**Comment by jbabyhacker** [comment]
I have tested it and it now works.
